slack_bot_rag/chain.py

```diff
@@ -0,0 +1,90 @@
+from langchain.memory import ConversationBufferMemory, DynamoDBChatMessageHistory
+from langchain.retrievers import AmazonKendraRetriever
+from langchain.chains import ConversationalRetrievalChain
+from langchain.chains import RetrievalQA
+
+from langchain.prompts import PromptTemplate
+from langchain import ConversationChain
+from langchain.prompts import (
+    ChatPromptTemplate, 
+    MessagesPlaceholder, 
+    SystemMessagePromptTemplate, 
+    HumanMessagePromptTemplate
+)
+from langchain.chat_models import ChatOpenAI
+import sys
+
+import config
+
+MAX_HISTORY_LENGTH = 5
+
+def run(api_key: str, session_id: str, prompt: str) -> str:
+    """This is the main function that executes the prediction chain.
+    Updating this code will change the predictions of the service.
+
+    Args:
+        api_key: api key for the LLM service, OpenAI used here
+        session_id: session id key to store the history
+        prompt: prompt question entered by the user
+
+    Returns:
+        The prediction from LLM
+    """
+    
+    chat_memory = DynamoDBChatMessageHistory(
+        table_name=config.config.DYNAMODB_TABLE_NAME,
+        session_id=session_id
+    )
+    
+    memory = ConversationBufferMemory(chat_memory=chat_memory, return_messages=True)   
+
+    retriever = AmazonKendraRetriever(index_id="d30dd38b-d307-4eba-90d0-c274639daf79")
```

Comment:
This should come in as an argument or an env variable. Please make sure, you don't put in your service ids in the PR here. 
```suggestion
    retriever = AmazonKendraRetriever(index_id=index_id)
```